fluidkeys/secretreceive.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/fluidkeys/fluidkeys/humanize"
+
+	"github.com/fluidkeys/fluidkeys/colour"
+
+	"github.com/fluidkeys/crypto/openpgp"
+	"github.com/fluidkeys/crypto/openpgp/armor"
+	"github.com/fluidkeys/fluidkeys/api"
+	"github.com/fluidkeys/fluidkeys/out"
+	"github.com/fluidkeys/fluidkeys/pgpkey"
+)
+
+func secretReceive() exitCode {
+	out.Print("\n")
+	client := api.NewClient()
+	keys, err := loadPgpKeys()
+
+	if err != nil {
+		printFailed("Couldn't load PGP keys")
+		return 1
+	}
+
+	out.Print(colour.Info("Downloading secrets...") + "\n\n")
+
+	for _, key := range keys {
+		if err != nil {
+			printFailed("Couldn't get email for key " + key.Fingerprint().String() + "\n")
+			continue
+		}
+		encryptedSecrets, err := client.ListSecrets(key.Fingerprint())
+		if err != nil {
+			out.Print("📪 " + displayName(&key) + ": " + colour.Failure(err.Error()) + "\n")
+			continue
+		}
+
+		if len(encryptedSecrets) == 0 {
+			out.Print("📭 " + displayName(&key) + ": No secrets found\n")
+			continue
+		}
+		var passwordPrompter promptForPasswordInterface
+		passwordPrompter = &interactivePasswordPrompter{}
+		if privateKey, _, err := getDecryptedPrivateKeyAndPassword(&key, passwordPrompter); err != nil {
+			err := fmt.Sprintf("Error getting private key and password: %s", err)
+			out.Print("📪 " + displayName(&key) + ": " + colour.Failure(err) + "\n")
+			continue
+		} else {
+			key = *privateKey
+		}
+
+		out.Print("📬 " + displayName(&key) + ":\n")
+
+		errors := []error{}
+		for i, encryptedSecret := range encryptedSecrets {
+			decryptedContent, err := decrypt(encryptedSecret.EncryptedContent, &key)
+			if err != nil {
+				errors = append(errors, err)
+			} else {
+				displayCounter := strconv.Itoa(i+1) + ". "
+				barLength := secretDividerLength - len(displayCounter)
+				out.Print(displayCounter + strings.Repeat("─", barLength) + "\n")
+
+				scanner := bufio.NewScanner(strings.NewReader(decryptedContent))
+				for scanner.Scan() {
```

Comment:
I'm not sure at all what's going on here!

Reply:
I've read the docs and I'm still not sure. My best guess: is it prepending a space before every line?